### ISSUE TITLE
Parse error: syntax error, unexpected 'protected' (T_PROTECTED), expecting variable (T_VARIABLE) in   vendor/collab/module-consent-mode/ViewModel/ModeHandler.php on line 20

### ISSUE THREAD

**Issue by lzss433** [collabpl/magento2-module-consent-mode#4]
magento version 2.3.7-p4
php version: 7.4.33

**Comment by kestraly** [comment]
@lzss433 Looks to be a bit of a mess. To fix for you

```
public function __construct(
       Config $config
    ) {
       $this->config = $config;
    }
```

**Comment by jemoon** [comment]
> magento version 2.3.7-p4 
> php version: 7.4.33

This module requires php 8.1+ - that's why You are getting this error - You can try if it will work on [feature/php7.4-compatibility](https://github.com/collabpl/magento2-module-consent-mode/tree/feature/php7.4-compatibility) branch but I don't have any plans of supporting this version so I won't fix anything if it won't work :/

@lzss433 I'm afraid that it won't be only thing which will require fixes in order to make it work under 7.4
